lib/ansible/modules/packaging/language/pip.py

```diff
@@ -457,16 +457,19 @@ def main():
                     if '--no-site-packages' in cmd_opts:
                         cmd += ' --no-site-packages'
 
-                if virtualenv_python:
-                    cmd += ' -p%s' % virtualenv_python
-                elif PY3:
-                    # Ubuntu currently has a patch making virtualenv always
-                    # try to use python2.  Since Ubuntu16 works without
-                    # python2 installed, this is a problem.  This code mimics
-                    # the upstream behaviour of using the python which invoked
-                    # virtualenv to determine which python is used inside of
-                    # the virtualenv (when none are specified).
-                    cmd += ' -p%s' % sys.executable
+                # -p is a virtualenv option, not compatible with pyenv or venv
+                # this if validates if the command being used is not any of them 
+                if not any((ex in module.params['virtualenv_command'] for ex in ['pyvenv', '-m venv'])):
+                    if virtualenv_python:
+                        cmd += ' -p%s' % virtualenv_python
+                    elif PY3:
+                        # Ubuntu currently has a patch making virtualenv always
+                        # try to use python2.  Since Ubuntu16 works without
+                        # python2 installed, this is a problem.  This code mimics
+                        # the upstream behaviour of using the python which invoked
+                        # virtualenv to determine which python is used inside of
+                        # the virtualenv (when none are specified).
+                        cmd += ' -p%s' % sys.executable
```

Comment:
Need the second ```elif: ``` from my example so that we throw an error if virtualenv_command is either pyvenv or -m venv